pkg/operator/ceph/object/spec.go

```diff
@@ -176,7 +238,26 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) v1.Container {
 		mount := v1.VolumeMount{Name: certVolumeName, MountPath: certDir, ReadOnly: true}
 		container.VolumeMounts = append(container.VolumeMounts, mount)
 	}
-
+	if c.clusterSpec.Security.KeyManagementService.IsEnabled() {
+		container.Args = append(container.Args,
+			cephconfig.NewFlag("rgw crypt s3 kms backend",
+				c.clusterSpec.Security.KeyManagementService.ConnectionDetails[authProvider]),
+			cephconfig.NewFlag("rgw crypt vault addr",
+				c.clusterSpec.Security.KeyManagementService.ConnectionDetails[vaultAddr]),
+			cephconfig.NewFlag("rgw crypt vault prefix",
+				c.clusterSpec.Security.KeyManagementService.ConnectionDetails[vaultBackendPath]),
+		)
+		if c.clusterSpec.Security.KeyManagementService.IsTokenAuthEnabled() {
+			container.VolumeMounts = append(container.VolumeMounts, c.vaultVolumeMount())
+			container.Args = append(container.Args,
+				cephconfig.NewFlag("rgw crypt vault auth", vaultKeyName),
+				cephconfig.NewFlag("rgw crypt vault token file",
+					path.Join(c.DataPathMap.ContainerDataDir, vaultFilename)),
+				// TODO: Make it configurable to include transit engine as well
```

Comment:
Let' use something like `VAULT_BACKEND` for this.

Reply:
You don't need a CRD setting, we can just read the var mentioned above in the KMS ConnectionDetails